pkg/flash/flash_linux.go

```diff
@@ -54,16 +60,36 @@ func New(spi SPI) (*Flash, error) {
 
 	var err error
 	f.sfdp, err = sfdp.Read(f.SFDPReader())
-	if err != nil {
-		return nil, fmt.Errorf("could not read sfdp: %v", err)
+	if err == nil {
+		if err = f.FillFromSFDP(); err == nil {
+			return f, nil
+		}
 	}
 
+	id, e := f.spi.ID()
+	if e != nil {
+		return nil, errors.Join(err, e)
+	}
+
+	if f.chip, e = chips.New(id); e != nil {
```

Comment:
I think it would be better to check for the Id before attempting to get the SFDP. The chip database might have more accurate information.

Reply:
I decided to try it, and ignore the error, let it fall through to trying the chip id.